plugin/storage/badger/spanstore/reader.go

```diff
@@ -237,8 +237,11 @@ func (r *TraceReader) GetServices(ctx context.Context) ([]string, error) {
 }
 
 // GetOperations fetches operations in the service and empty slice if service does not exists
-func (r *TraceReader) GetOperations(ctx context.Context, service string) ([]string, error) {
-	return r.cache.GetOperations(service)
+func (r *TraceReader) GetOperations(
+	ctx context.Context,
+	query *spanstore.OperationQueryParameters,
+) ([]*spanstore.Operation, error) {
```

Comment:
before it's too late, I suggest changing this to return slice of values `[]spanstore.Operation`. There's zero benefit in returning pointers, only extra allocations.

Reply:
I can also change the parameter so the signature will be:

GetOperations(ctx context.Context, query OperationQueryParameters) ([]Operation, error)

I will not increase the scope of this PR by updating below methods:
```
FindTraces(ctx context.Context, query *TraceQueryParameters) ([]*model.Trace, error)
FindTraceIDs(ctx context.Context, query *TraceQueryParameters) ([]model.TraceID, error)
```

Reply:
ok